### ISSUE TITLE
[$1000] Marking Concierge's message as unread doesn't work when it's the last message

### ISSUE THREAD

**Issue by kavimuru** [Expensify/App#20109]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:
1. Open any report
2. Flag a message as offensive
3. When Concierge's message shows up, hover over it
4. Click on Mark as unread icon
## Expected Result: 
Concierge's message is marked as unread even if it's the last message
## Actual Result: 
Concierge's message is not marked as unread if it's the last message. It's marked as unread if it's not the last message

## Workaround:
Can the user still use Expensify without this being fixed? Have you informed them of the workaround?

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [x] Android / native
- [x] Android / Chrome
- [x] iOS / native
- [x] iOS / Safari
- [x] MacOS / Chrome / Safari
- [x] MacOS / Desktop

**Version Number:** 1.3.22-0
**Reproducible in staging?:** y
**Reproducible in production?:** y
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Notes/Photos/Videos:** Any additional supporting documentation

https://github.com/Expensify/App/assets/43996225/ad8689c8-5509-4675-9617-2d2d3a58789d

https://github.com/Expensify/App/assets/43996225/6c0fe209-9311-4c45-af16-5c06336af8be

**Expensify/Expensify Issue URL:**
**Issue reported by:** @adeel0202 
**Slack conversation:** https://expensify.slack.com/archives/C049HHMV9SM/p1685635547475619

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~01e7489bcbd82f8d62</li>
        <li>Upwork Job ID: 1665724453061107712</li>
        <li>Last Price Increase: 2023-06-19</li>
    </ul>
</details>

**Comment by namhihi237** [comment]
This bug is not frequent I sometimes get

**Comment by namhihi237** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
Concierge's message is marked as unread even if it's the last message
### What is the root cause of that problem?
Step shows error: 
1. Open a report 
2. Flag message 
3. Click another report and open the previous report
4.  Mark as unread

This is the report_id that is obtained in Onyx:
<img width="962" alt="Screenshot 2023-06-03 at 22 30 50" src="https://github.com/Expensify/App/assets/39086067/06973933-867e-4a70-813e-b764ceb762c2">

We can see (above image) after step 3 when reopening the report, the data returned to the report does not have the message "Thanks for...." it has returned the previous user's message.
API mark unread when sending `lastReadTime`: 
`E4 = lastReadTime = DateUtils.subtractMillisecondsFromDateTime(reportActionCreated, 1);`
reportActionCreated = C5
This leads to berth `lastReadTime` > `lastVisibleActionCreated`
And this code used to check marks always returns false because `ReportUtils.isUnread(this.props.report) = false `
https://github.com/Expensify/App/blob/0813e2ca1476635a8324cbd9f998424663a33128/src/pages/home/report/ReportActionsView.js#L218-L223
### What changes do you think we should make in order to solve the problem?
I think we should check the API open report why the data is wrong when there is Concierge's message in the last.
### What alternative solutions did you explore? (Optional)
N/A

**Comment by flaviadefaria** [comment]
Added the external label

**Comment by flaviadefaria** [comment]
@fedirjh any thoughts on the proposal above?

**Comment by marcaaron** [comment]
@fedirjh I'm not really sure tbh. I think the expected behavior looks reasonable to me as the "system message" is not distinguishable from any other kind of comment. Why it is not being considered the `lastVisibleActionCreated` is  a total mystery to me.

@dangrous do you have any thoughts on this?

**TL;DR on the bug:**

The messages in the chat that pop up after someone flags something as offensive can't be "marked as unread"

**Comment by pecanoro** [comment]
All the lastVisibleAction* is not returned when the last message is a whisper. It is a technical limitation we discussed in Slack when this was implemented. When these actions get added to the report, we don't have a way to get the last action as it depends on which user is seeing the report. Since we usually notify everyone at the same time, we would need a new command or something that gets the last message for all participants. Because of that, we default it to the last non-whisper action. I don't know if this has any solution at all, when this was discussed, we mentioned the whole lastMessage system for the LHN would need a refactor to make it work.

**Comment by Victor-Nyagudi** [comment]
@fedirjh I think this issue could be a potential candidate for being added to the [tracking issue for problems with the new marker line](https://github.com/Expensify/App/issues/15212).

cc: @pecanoro @dangrous @marcaaron

**Comment by flaviadefaria** [comment]
@fedirjh do you have an update here? Any good proposals so far?

**Comment by flaviadefaria** [comment]
Got it thanks for clarifying @fedirjh. @pecanoro should we just close this then?

**Comment by flaviadefaria** [comment]
Closing this based on @pecanoro's comment. Please reopen if this needs to be worked on internally but seems like it would require a major change and so probably a bigger project.
